src/fighter_state.rs

```diff
@@ -0,0 +1,747 @@
+        }
+    }
+}
+
+/// Handle fighter moving state
+fn moving(
+    mut commands: Commands,
+    mut fighters: Query<(
+        Entity,
+        &mut Animation,
+        &mut Facing,
+        &mut LinearVelocity,
+        &Moving,
+    )>,
+) {
+    for (entity, mut animation, mut facing, mut velocity, moving) in &mut fighters {
+        // If we aren't playing the moving animation
+        if animation.current_animation.as_deref() != Some(Moving::ANIMATION) {
+            // Start the moving animation from the beginning
+            animation.play(Moving::ANIMATION, true /* repeating */);
+        }
+
+        // Update our velocity to match our movement velocity
+        **velocity = moving.velocity;
+
+        // Make sure we face in the direction we are moving
+        if velocity.x > 0.0 {
+            *facing = Facing::Right
+        } else if velocity.x < 0.0 {
+            *facing = Facing::Left
+        }
+
+        // Moving is a little different than the other states because we transition out of it at the
+        // end of every frame, so that we only move if the player continually inputs a movement.
+        commands.entity(entity).remove::<Moving>().insert(Idling);
+    }
+}
+
+/// Update knocked back players
+fn knocked_back(
+    mut fighters: Query<(
+        &mut Animation,
+        &Facing,
+        &mut LinearVelocity,
+        &mut KnockedBack,
+    )>,
+    time: Res<Time>,
+) {
+    for (mut animation, facing, mut velocity, mut knocked_back) in &mut fighters {
+        // If this is the start of the knock back
+        if knocked_back.timer.elapsed_secs() == 0.0 {
+            // Calculate animation to use based on attack direction
+            let is_left = knocked_back.velocity.x < 0.0;
+            let use_left_anim = if facing.is_left() { !is_left } else { is_left };
+            let animation_name = if use_left_anim {
+                KnockedBack::ANIMATION_LEFT
+            } else {
+                KnockedBack::ANIMATION_RIGHT
+            };
+
+            // Play the animation
+            animation.play(animation_name, false);
+        }
+
+        // Tick the knock-back timer
+        knocked_back.timer.tick(time.delta());
+
+        // Set our figher velocity to the knock back velocity
+        **velocity = knocked_back.velocity;
+    }
+}
+
+/// Update dying players
+fn dying(
+    mut commands: Commands,
+    mut fighters: Query<(Entity, &mut Animation, &mut LinearVelocity), With<Dying>>,
+) {
+    for (entity, mut animation, mut velocity) in &mut fighters {
+        // Start playing the dying animation if it isn't already
+        if animation.current_animation.as_deref() != Some(Dying::ANIMATION) {
+            **velocity = Vec2::ZERO;
+            animation.play(Dying::ANIMATION, false);
+
+        // When the animatino is finished, despawn the fighter
+        } else if animation.is_finished() {
+            commands.entity(entity).despawn_recursive();
+        }
```

Comment:
In regards to our `DespawnMarker` discussion, I'm thinking that if `dying` is the highest priority state and exclusive, and no other states despawn fighters we will not have the commands applied to non-existent entity issue at least for Fighters. Could still be an issue for items or other things but unsure.

Reply:
OK, then I think, since it's so easy to implement later, if necessary, we can just wait to see if we run into issues.